webapp/views.py

```diff
@@ -795,10 +795,22 @@ def make_renewal(advantage, contract_info):
 
     sorted_renewals = sorted(
         renewals,
-        key=lambda renewal: dateutil.parser.parse(renewal["start"]),
+        key=lambda renewal: dateutil.parser.parse(
+            renewal.get("start", "9999-12-12T00:00:00Z")
+        ),
     )
 
-    renewal = sorted_renewals[0]
+    # Check for the status key before returning a renewal. It is possible
+    # that it is not present, which would indicate an invalid renewal.
+    renewal = None
+    for r in sorted_renewals:
+        # Return first renewal with a status present.
+        if "status" in r:
+            renewal = r
+            break
+
+    if not renewal:
```

Comment:
Why not just return renewal? If it's None it will return None.

Reply:
The dict lookups below will error if renewal is None. 

In any case, I'm going to remove this and just wrap the whole thing in a KeyError check in a few minutes when I get back to my keyboard

Reply:
Oh sorry of course, the diff cut off the rest of the function.